pykafka/cli/kafka_tools.py

```diff
@@ -0,0 +1,285 @@
+    for owned_partition, partition in consumer._partitions.iteritems():
+        owned_partition.last_offset_consumed = int(offsets[partition.id].offset[0])
+    consumer.commit_offsets()
+
+
+def _get_arg_parser():
+    output = argparse.ArgumentParser(description='Tools for Kafka.')
+
+    # Common arguments
+    output.add_argument('-b', '--broker',
+                        required=False,
+                        default='localhost:9092',
+                        dest='host',
+                        help='host:port of any Kafka broker. '
+                             '[default: localhost:9092]')
+
+    subparsers = output.add_subparsers(help='Commands', dest='command')
+
+    # Desc Topic
+    parser = subparsers.add_parser(
+        'desc_topic',
+        help='Print detailed info for a topic.'
+    )
+    parser.add_argument('topic',
+                        metavar='TOPIC',
+                        help='Topic name.')
+
+    # Print Consumer Lag
+    parser = subparsers.add_parser(
+        'print_consumer_lag',
+        help='Get consumer lag for a topic.'
+    )
+    parser.add_argument('topic',
+                        metavar='TOPIC',
+                        help='Topic name.')
+    parser.add_argument('consumer_group',
+                        metavar='CONUSMER_GROUP',
+                        help='Consumer group name.')
+
+    # Print Offsets
+    parser = subparsers.add_parser(
+        'print_offsets',
+        help='Fetch offsets for a topic/consumer group'
+    )
+    parser.add_argument('topic',
+                        metavar='TOPIC',
+                        help='Topic name.')
+    parser.add_argument('offset',
+                        metavar='OFFSET',
+                        type=str,
+                        help='Offset to fetch. Can be EARLIEST, LATEST, or a '
+                             'datetime in the format YYYY-MM-DDTHH:MM:SS.')
+
+    # Print Topics
+    parser = subparsers.add_parser(
+        'print_topics',
+        help='Print information about all topics in the cluster.'
+    )
+
+
+    # Reset Offsets
+    parser = subparsers.add_parser(
+        'reset_offsets',
+        help='Reset offsets for a topic/consumer group'
+    )
+    parser.add_argument('topic',
+                        metavar='TOPIC',
+                        help='Topic to reset offsets for.')
+    parser.add_argument('consumer_group',
+                        metavar='CONUSMER_GROUP',
+                        help='Consumer group to reset offsets for.')
+    parser.add_argument('offset',
+                        metavar='OFFSET',
+                        type=str,
+                        help='Target offset. Can be EARLIEST, LATEST, or a '
+                             'datetime in the format YYYY-MM-DDTHH:MM:SS.')
+
+    return output
+
+
+def main():
+    parser = _get_arg_parser()
+    args = parser.parse_args()
+
+    # Connect to Kafka
+    client = pykafka.KafkaClient(hosts=args.host)
+
+    if args.command == 'reset_offsets':
```

Comment:
Instead of doing this to make the correct function get called, you can just do:

```python
parser.set_defaults(func=reset_offsets)
```

when you're creating the subparser, and make each of these function take `args` instead of individual arguments.

Then this whole if block can just become:

```python
args.func(args)
```